### PR TITLE
Add direct deposit logs

### PR DESCRIPTION
**Pull request by tpharrison**
## Summary

- Add temporary logging to controller.
- Refactored `control_information` to use `ActiveModel::Attributes` and `ActiveModel::AttributeAssignment`.  This helped with coercing request parameters.
- Added/updated specs.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/79250

## Testing done
Added/updated specs in:
- direct_deposits_controller_spec
- control_information_spec

### CODE REVIEW

**Review by adamwhitlock1 — APPROVED** [review]
lgtm

File: app/controllers/v0/profile/direct_deposits_controller.rb [hunk 1]
```diff
@@ -44,6 +50,15 @@ def update
 
       private
 
+      def log_control_info(control_info, action)
+        ::Rails.logger.info("Direct Deposit Control Info: #{action}",
+                            { benefit_type: control_info.benefit_type,
+                              updatable: control_info.account_updatable?,
+                              valid: control_info.valid?,
+                              restrictions: control_info.restrictions.join(', '),
+                              errors: control_info.errors.join(', ') })
```
Comment (rmtolmach) [inline app/controllers/v0/profile/direct_deposits_controller.rb:59]: It doesn't look like it, but I want to be sure... is there any PII in here? (I see "direct deposit" and I worry 😬)

File: app/controllers/v0/profile/direct_deposits_controller.rb [hunk 2]
```diff
@@ -84,7 +100,7 @@ def control_info_params
                       :has_no_fiduciary_assigned,
                       :is_not_deceased,
                       :has_payment_address,
-                      :has_indentity)
+                      :has_identity)
```
Comment (rmtolmach) [inline app/controllers/v0/profile/direct_deposits_controller.rb:103]: Nice catch!

File: app/controllers/v0/profile/direct_deposits_controller.rb (hunk 1 above)
Comment (tpharrison) [reply app/controllers/v0/profile/direct_deposits_controller.rb:59]: @rmtolmach No PII.  Thank you for checking though.  The control info fields are just boolean flags related to the user and account.   This extra logging is so we can monitor activity around some changes to direct deposit.  It'll most likely be temporary.

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**